nfsv4/transfer.c

```diff
@@ -10,7 +10,11 @@ status finish_read(void *dest, buffer b)
     u32 len = read_beu32(b);
     memcpy(dest, buffer_ref(b, 0), len);
     // assumes ordering
-    b->start += len;
+
+    // data is padded into multiple of 4
+    // round length up to nearest multiple of 4
```

Comment:
See the comment on lines 3-5 above.  This seems related to the problem you are addressing and should probably be fixed.

Reply:
ah yes - this is the last fragment case